subsys/senss/include/senss_sensor.h

```diff
@@ -0,0 +1,605 @@
+ * @param dev The sensor instance device structure.
+ *
+ * @param buf The data buffer.
+ *
+ * @param size The buffer size in bytes.
+ *
+ * @return 0 on success or negative error value on failure.
+ */
+int senss_sensor_post_data(
+		const struct device *dev,
+		void *buf, int size);
+
+/**
+ * @brief Get reporter handles	of a given sensor instance by sensor type.
+ *
+ * @param dev The sensor instance device structure.
+ *
+ * @param type The given type, \ref SENSS_SENSOR_TYPE_ALL to get reporters
+ * with all types.
+ *
+ * @param max_handles The max count of the \p reporter_handles array input. Can
+ * get real count number via \ref senss_sensor_get_reporters_count
+ *
+ * @param *reporter_handles Input handles array for receiving found reporter
+ * sensor instances
+ *
+ * @return number of reporters found, 0 returned if not found.
+ */
+int senss_sensor_get_reporters(
+		const struct device *dev, int type,
+		const senss_sensor_handle_t *reporter_handles, int max_handles);
+
+/**
+ * @brief Get reporters count of a given sensor instance by sensor type.
+ *
+ * @param dev The sensor instance device structure.
+ *
+ * @param type The sensor type for checking, \ref SENSS_SENSOR_TYPE_ALL
+ *
+ * @return Count of reporters by \p type, 0 returned if no reporters by \p type.
+ */
+int senss_sensor_get_reporters_count(
+		const struct device *dev, int type);
+
+/**
+ * @brief Get this sensor's state
+ *
+ * @param dev The sensor instance device structure.
+ *
+ * @param state Returned sensor state value
+ *
+ * @return 0 on success or negative error value on failure.
+ */
+int senss_sensor_get_state(
+		const struct device *dev,
+		enum senss_sensor_state *state);
+
+
+/**
+ * @}
+ */
+
+/**
+ * @brief Sensor Callbacks
+ * @addtogroup senss_sensor_callbacks
+ * \{
+ */
+
+/**
+ * @brief Sensor initialize callback for virtual sensor.
+ *
+ * Sensor can initialize it's runtime context in this callback.
+ *
+ * @param dev The sensor instance device structure.
+ *
+ * @param info The sensor instance's constant information.
+ *
+ * @param reporter_handles The reporters handles for this sensor.
+ *
+ * @param reporter_count The number of reporters.
+ *
+ * @return 0 on success or negative error value on failure.
+ *
+ */
+typedef int (*senss_sensor_virtual_init_t)(
+		const struct device *dev, const struct senss_sensor_info *info,
+		const senss_sensor_handle_t *reporter_handles, int reporters_count);
+
+/**
+ * @brief Sensor initialize callback for physical sensor.
+ *
+ * Sensor can initialize it's runtime context in this callback.
+ *
+ * @param dev The sensor instance device structure.
+ *
+ * @param info The sensor instance's constant information.
+ *
+ * @param underlying_dev The underlying sensor device from kernel space.
+ *
+ * @return 0 on success or negative error value on failure.
+ *
+ */
+typedef int (*senss_sensor_physical_init_t)(
+		const struct device *dev, const struct senss_sensor_info *info,
+		const struct device *underlying_dev);
```

Comment:
Also discussed in a meeting so I'm adding this comment here for visibility.

This should be the same as the virtual sensor. The info should be returned from the sensor, not populated by the subsystem. Also, the underlying dev can be attained by the sensor driver via `DEVICE_DT_GET` and set in the device's `config` at link time instead of at run time.

Reply:
Yes, now, the physical sensor API and virtual sensor API have combined to single API set. see https://github.com/zephyrproject-rtos/zephyr/pull/45370#discussion_r893421127

Now, we used same `init` API for all sensors: 

```c
static struct acc_context base_acc_ctx = {
   .underlying_dev = DEVICE_DT_GET(DT_PHANDLE(DT_NODELABEL(base_accel), underlying_device))
};

static struct acc_context lid_acc_ctx = {
   .underlying_dev = DEVICE_DT_GET(DT_PHANDLE(DT_NODELABEL(lid_accel), underlying_device))
};

int acc_init(const struct device *dev, const struct senss_sensor_info *info,
    const senss_sensor_handle_t *reporter_handles, int reporters_count)
{
    printf("Sensor instance name: %s\n", dev->name);

    /* reporter_handles should be NULL,  and reporters_count should be zero */

    struct acc_context *ctx = senss_sensor_get_ctx_data(dev);
    printf("Underlying device: %s\n", ctx->underlying_dev->name);

    return 0;
}

```
The example is in document: https://github.com/huhebo/zephyr/blob/subsys/sensorsubsys/doc/services/senss/index.rst